File: server/datasets/java_source-code/Multiple-G1/renamed_42_bst.java
class Node_6 {
    int val_6;
    Node_6 lhs_6, rhs_6;
    public Node_6(int item) {
        val_6 = item;
        lhs_6 = rhs_6 = null;
    }
}
public class Tree_6 {
    public static void main(String[] args) {
        Tree_6 bst = new Tree_6();
        bst.ins_6(50);
        bst.srch_6(50);
    }
    Node_6 rt_6;
    public void ins_6(int val_6) {
        rt_6 = in_rec_6(rt_6, val_6);
    }
    Node_6 in_rec_6(Node_6 rt_6, int val_6) {
        if (rt_6 == null) return new Node_6(val_6);
        if (val_6 < rt_6.val_6) rt_6.lhs_6 = in_rec_6(rt_6.lhs_6, val_6);
        else if (val_6 > rt_6.val_6) rt_6.rhs_6 = in_rec_6(rt_6.rhs_6, val_6);
        return rt_6;
    }
    Node_6 srch_6(int val_6) {
        return sr_rec_6(rt_6, val_6);
    }
    Node_6 sr_rec_6(Node_6 rt_6, int val_6) {
        if (rt_6 == null || rt_6.val_6 == val_6) return rt_6;
        if (rt_6.val_6 < val_6) return sr_rec_6(rt_6.rhs_6, val_6);
        return sr_rec_6(rt_6.lhs_6, val_6);
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g1/renamed_42_bst.py
class Node_6:
    def __init__(self, val_6):
        self.val_6 = val_6
        self.lhs_6 = None
        self.rhs_6 = None

class Tree_6:
    def __init__(self):
        self.rt_6 = None

    def ins_6(self, val_6):
        if self.rt_6 is None:
            self.rt_6 = Node_6(val_6)
        else:
            self._ins_6_rec(self.rt_6, val_6)

    def _ins_6_rec(self, nd_6, val_6):
        if val_6 < nd_6.val_6:
            if nd_6.lhs_6 is None:
                nd_6.lhs_6 = Node_6(val_6)
            else:
                self._ins_6_rec(nd_6.lhs_6, val_6)
        elif val_6 > nd_6.val_6:
            if nd_6.rhs_6 is None:
                nd_6.rhs_6 = Node_6(val_6)
            else:
                self._ins_6_rec(nd_6.rhs_6, val_6)

    def srch_6(self, val_6):
        return self._srch_6_rec(self.rt_6, val_6)

    def _srch_6_rec(self, nd_6, val_6):
        if nd_6 is None or nd_6.val_6 == val_6:
            return nd_6
        if val_6 < nd_6.val_6:
            return self._srch_6_rec(nd_6.lhs_6, val_6)
        return self._srch_6_rec(nd_6.rhs_6, val_6)
